pkg/middlewares/google_compute_instance_group_manager_expander.go

```diff
@@ -0,0 +1,47 @@
+package middlewares
+
+import (
+	"github.com/snyk/driftctl/pkg/resource"
+	"github.com/snyk/driftctl/pkg/resource/google"
+)
+
+type GoogleComputeInstanceGroupManagerExpander struct{}
+
+func NewGoogleComputeInstanceGroupManagerExpander() *GoogleComputeInstanceGroupManagerExpander {
+	return &GoogleComputeInstanceGroupManagerExpander{}
+}
+
+func (a GoogleComputeInstanceGroupManagerExpander) Execute(remoteResources, resourcesFromState *[]*resource.Resource) error {
+	var newStateResources []*resource.Resource
+
+	instanceGroups := make([]*resource.Resource, 0)
+	for _, remoteResource := range *remoteResources {
+		// Ignore all resources other than google_compute_instance_group
+		if remoteResource.ResourceType() != google.GoogleComputeInstanceGroupResourceType {
+			continue
+		}
+		instanceGroups = append(instanceGroups, remoteResource)
+	}
+
+	for _, stateResource := range *resourcesFromState {
+		newStateResources = append(newStateResources, stateResource)
+
+		// Ignore all resources other than google_compute_instance_group
+		if stateResource.ResourceType() != google.GoogleComputeInstanceGroupManagerResourceType {
+			continue
+		}
+
+		name := stateResource.Attributes().GetString("name")
+
+		for _, group := range instanceGroups {
+			// Import instance group in the state
+			if n := group.Attributes().GetString("name"); n != nil && *n == *name {
+				newStateResources = append(newStateResources, group)
```

Comment:
You are already appending resource in the state L27, I'm not sure to understand the logic here

Reply:
What I do at L27 is to add each state resource in the new array. L39 adds the remote instance group in the state. Is it clear enough?

Reply:
> Is it clear enough?

Absolutely not 😅 
Could you add a very complete explanation for someone that do not know nothing about that particular resource as I asked here https://github.com/snyk/driftctl/pull/1379#discussion_r834092080

Reply:
Added a comment, hope it's clearer now, let me know

Reply:
He's just matching `google_compute_instance_group_manager` name with `google_compute_instance_group_manager` name to find which ones are managed by the group manager.

I would have named it `google_compute_instance_group_manager_reconciler`.

And don't forget to change the name of the file.